backend/amd64/selection.ml

```diff
@@ -298,53 +298,19 @@ method! select_operation op args dbg =
       self#select_floatarith true width Imulf Ifloatmul args
   | Cdivf width ->
       self#select_floatarith false width Idivf Ifloatdiv args
-  (* Special cases overriding C implementations. *)
-  | Cextcall { func = "sqrt"; alloc = false; } ->
-     begin match args with
-       [Cop(Cload { memory_chunk = Double as chunk; _}, [loc], _dbg)] ->
-         let (addr, arg) = self#select_addressing chunk loc in
-         (Ispecific(Ifloatsqrtf (Float64, addr)), [arg])
-     | [arg] ->
-         (Ispecific Simd.(Isimd (SSE2 Sqrt_scalar_f64)), [arg])
-     | _ ->
-         assert false
-    end
-  | Cextcall { func = "caml_int64_bits_of_float_unboxed"; alloc = false;
-               ty = [|Int|]; ty_args = [XFloat] } ->
-      (match args with
-        | [Cop(Cload { memory_chunk = Double; mutability = mut; is_atomic }, [loc], _dbg)] ->
-        let c = Word_int in
-        let (addr, arg) = self#select_addressing c loc in
-        Iload { memory_chunk = c;
-                addressing_mode = addr;
-                mutability = mut;
-                is_atomic; }, [arg]
-      | _ -> Imove, args)
-  | Cextcall { func = "caml_int64_float_of_bits_unboxed"; alloc = false;
-               ty = [|Float|]; ty_args = [XInt64] } ->
-      (match args with
-      | [Cop(Cload { memory_chunk = Word_int; mutability = mut; is_atomic }, [loc], _dbg)] ->
-        let c = Double in
-        let (addr, arg) = self#select_addressing c loc in
-        Iload { memory_chunk = c;
-                addressing_mode = addr;
-                mutability = mut;
-                is_atomic; }, [arg]
-      | _ -> Imove, args)
+  (* Special cases overriding C implementations (regardless of [@@builtin]). *)
+  | Cextcall { func = ("sqrt" as func); _ }
```

Comment:
Don't we also want to catch `"sqrtf"`?

Reply:
`"sqrtf"` is allowed in `select_operation_sse2` - that means the intrinsic requires `[@@builtin]`, but I think that's fine since it didn't already exist in the stdlib

Reply:
There is an `external` declaration for `sqrt` in `stdlib` and its compiled directly to `sqrt` call to libc for native code.  If we put `builtin` and require it to go through sse2, wouldn't that version be ignored and the old one used instead? Can you make a test to check that both ways work?

Reply:
I wasn't planning on adding `builtin` to the `sqrt` in stdlib/base/core, since this case still applies even without it.
Is that fine?

Reply:
@TheNumbat I was referring to your plan on having `sqrt` in `sse2` selection with `builtin`, I understand that you are not adding `@@builtin` in stdlib.  You said above

> I think that's fine since it didn't already exist in the stdlib

but it does exist in stdlib, and it doesn't have `builtin` there, so there will be two places where `sqrt` intrinsics is translated to an instructon.

Reply:
(Discussed with Greta offline)